srcs/incs/ConnectionIOStatus.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef SRCS_INCS_CONNECTIONIOSTATUS_HPP_
+#define SRCS_INCS_CONNECTIONIOSTATUS_HPP_
+
+struct ReadRequestStatus {
+	enum Type {
+		kFail,
+		kSuccess,
+		kStart
+	};
```

Comment:
Would you like the idea of having a `typedef` here?

```suggestion
	typedef enum Type {
		kFail,
		kSuccess,
		kStart
	} Type;
```

Reply:
I like this, added.